include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -754,6 +758,74 @@ __select_backend(const execution::fpga_policy<_Factor, _KernelName>&, _Ranges&&.
 }
 #endif
 
+// Check the outer view type type to see if we can vectorize. Any non-contiguous inputs (e.g. reverse
+// views, permutation views, etc.) cannot be vectorized. If C++20 ranges are present, then we can
+// use the std::ranges::contiguous_range concept.
+template <typename _Rng>
+struct __is_vectorizable_range
+{
+    constexpr static bool value =
+#if _ONEDPL_CPP20_RANGES_PRESENT && _ONEDPL_CPP20_CONCEPTS_PRESENT
+        std::ranges::contiguous_range<_Rng> ||
+#endif
+        __nanorange::nano::ranges::contiguous_range<_Rng>;
+};
+
+// Guard view specializations
+template <typename _Type>
+struct __is_vectorizable_range<oneapi::dpl::__ranges::guard_view<_Type*>> : std::true_type
```

Comment:
`std::is_pointer<>` is better

```
template <typename _Type>
struct __is_vectorizable_range<oneapi::dpl::__ranges::guard_view<_Type>> : std::is_pointer<_Type>
```

Reply:
I switched to `std::is_pointer`. However, the suggested approach does not work as it would make `__is_vectorizable_range` false for all guard views over iterators that are not pointers. In practice it causes redefinition compiler errors.

What I switched to is this:
```
template <typename _Pointer>
struct __is_vectorizable_range<oneapi::dpl::__ranges::guard_view<_Pointer>,
                               std::enable_if_t<std::is_pointer_v<_Pointer>>> : std::true_type
{
};
```
with a defaulted `typename = void` template parameter in `__is_vectorizable_range` The general iterator case also requires a `std::enable_if_t<!std::is_pointer_v<_Pointer>>` to disambiguate.